docs/proposals/block-storage-time-series-deletion.md

```diff
@@ -0,0 +1,272 @@
+
+Similar to the chunk storage deletion implementation, the initial filtering of the deleted data will be done inside the Querier. This will allow filtering the data read from both the store gateway and the ingestor. This functionality already exists for the chunk storage implementation. By implementing it in the querier, this would mean that the ruler can also utilize this to filter out the various metrics for the alert manager (read from the store gateway).  
+
+
+#### Storing tombstones in object store 
+
+
+The Purger will store the tombstone entries in a separate folder called “series_deletion” in the object store (e.g. S3 bucket) in the respective tenant folder. Each tombstone can have a separate JSON file outlining all the necessary information about the deletion request such as the parameters passed in the request, as well as the current status and some meta-data such as the creation date of the request.  The name of the file can be a hash of the API parameters (start, end, markers). This way if a user calls the API twice by accident with the same parameters, it will only create one tombstone.  
+
+The tombstone will be stored in a single file per request: 
+
+- `/<tenantId>/series_deletion/requests/<request_id>.json`
+
+
+The schema of the JSON file is:
+
+
+```
+{
+  "requestId": <string>,
+  "startTime": <int>,
+  "endTime": <int>,
+  "creationTime": <int>,
+  "matchers": [
+    "<string matcher 1>",
+    .., 
+    "<string matcher n>"
+    ]
+  },
+  "userID": <string>,
+  "deleteRequestStatus": <received | buildingPlan | deleting | processed> 
+}
+```
+
+
+Pros:
+
+- Design is similar to the existing chunk storage deletion 
+    - Lots of code can be reused inside the purger component. 
+- Allows deletion and un-delete to be done in a single operation. 
+
+Cons:
+- Negative impact on query performance when there are active tombstones. As in the chunk storage implementation, all the queries made will have to be compared to the matchers contained in the active tombstone files. The impact on performance should be the same as the deletion would have with chunk storage.
+
+
+#### Invalidating cache
+
+Using block store, the different caches available are:
+- Index cache
+- Metadata cache
+- Chunks cache (stores the potentially to be deleted chunks of data) 
+- Query results cache (stores the potentially to be deleted data) 
+
+The filtering using tombstones is only requested by the querier. By using the purger, the querier filters out the data received from the ingestors and store-gateway. The cache not being processed through the querier needs to be invalidated to prevent deleted data from coming up in queries. There are two potential caches that could contain deleted data, the chunks cache, and the query results cache. 
```

Comment:
> By using the purger, the querier filters out the data received from the ingestors and store-gateway.

I don't fully understand what do you mean with "by using the purger" in this context.

Reply:
Great catch, mistake on my part. It should have been tombstone information.